src/hotspot/cpu/x86/x86.ad

```diff
@@ -1753,15 +1751,23 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
     case Op_ClearArray:
     case Op_VectorMaskGen:
     case Op_VectorCmpMasked:
-    case Op_LoadVectorMasked:
-    case Op_StoreVectorMasked:
       if (!is_LP64 || !VM_Version::supports_avx512bw()) {
         return false;
       }
       if ((size_in_bits != 512) && !VM_Version::supports_avx512vl()) {
         return false;
       }
       break;
+    case Op_LoadVectorMasked:
+      if (!VM_Version::supports_avx512bw() && (is_subword_type(bt) || UseAVX < 1)) {
+        return false;
+      }
+      break;
+    case Op_StoreVectorMasked:
+      if (!VM_Version::supports_avx512bw() && (is_subword_type(bt) || UseAVX < 1)) {
```

Comment:
Why duplicate the same check?
And you did not answer my suggestion about modifying the check.